launcher/ui/pages/instance/LogPage.cpp

```diff
@@ -36,331 +36,288 @@
-    }
+    QString fontFamily = APPLICATION->settings()->get("ConsoleFont").toString();
+    bool conversionOk = false;
+    int fontSize = APPLICATION->settings()->get("ConsoleFontSize").toInt(&conversionOk);
+    if (!conversionOk)
+        fontSize = 11;
 
-    ui->text->setModel(m_proxy);
+    m_proxy = new LogFormatProxyModel(LogColorCache(origForeground, origBackground), QFont(fontFamily, fontSize), this);
 
     // set up instance and launch process recognition
     {
-        auto launchTask = m_instance->getLaunchTask();
-        if(launchTask)
-        {
+        if (auto launchTask = m_instance->getLaunchTask(); launchTask)
             setInstanceLaunchTaskChanged(launchTask, true);
-        }
         connect(m_instance.get(), &BaseInstance::launchTaskChanged, this, &LogPage::onInstanceLaunchTaskChanged);
     }
 
-    auto findShortcut = new QShortcut(QKeySequence(QKeySequence::Find), this);
-    connect(findShortcut, SIGNAL(activated()), SLOT(findActivated()));
-    auto findNextShortcut = new QShortcut(QKeySequence(QKeySequence::FindNext), this);
-    connect(findNextShortcut, SIGNAL(activated()), SLOT(findNextActivated()));
-    connect(ui->searchBar, SIGNAL(returnPressed()), SLOT(on_findButton_clicked()));
-    auto findPreviousShortcut = new QShortcut(QKeySequence(QKeySequence::FindPrevious), this);
-    connect(findPreviousShortcut, SIGNAL(activated()), SLOT(findPreviousActivated()));
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    // Signal never gets called in Qt5 for some god-forsaken reason o.O
+    onQuickWidgetReady();
+#endif
 }
 
-LogPage::~LogPage()
+bool LogPage::shouldDisplay() const
 {
-    delete ui;
+    return m_instance->isRunning() || m_proxy->rowCount() > 0;
 }
 
-void LogPage::modelStateToUI()
+bool LogPage::apply()
 {
-    if(m_model->wrapLines())
-    {
-        ui->text->setWordWrap(true);
-        ui->wrapCheckbox->setCheckState(Qt::Checked);
-    }
-    else
-    {
-        ui->text->setWordWrap(false);
-        ui->wrapCheckbox->setCheckState(Qt::Unchecked);
-    }
-    if(m_model->suspended())
-    {
-        ui->trackLogCheckbox->setCheckState(Qt::Unchecked);
-    }
-    else
-    {
-        ui->trackLogCheckbox->setCheckState(Qt::Checked);
-    }
+    saveSettings();
+
+    return true;
 }
 
-void LogPage::UIToModelState()
+void LogPage::onStatusChanged(QQuickWidget::Status status)
 {
-    if(!m_model)
-    {
-        return;
-    }
-    m_model->setLineWrap(ui->wrapCheckbox->checkState() == Qt::Checked);
-    m_model->suspend(ui->trackLogCheckbox->checkState() != Qt::Checked);
+    if (status == QQuickWidget::Status::Ready)
+        onQuickWidgetReady();
+}
+
+void LogPage::onQuickWidgetReady()
+{
+    [[maybe_unused]] bool no_errors = QQmlProperty::write(rootObject(), "logModel", QVariant::fromValue(m_proxy));
+    Q_ASSERT(no_errors);
+
+    no_errors = connect(rootObject(), SIGNAL(onWrapModeChanged(int)), this, SLOT(wrapModeChanged(int)));
+    Q_ASSERT(no_errors);
```

Comment:
This `Q_ASSERT(no_errors);` pattern seems like something that will need to be reused with every Qml object we want to connect. should we write a simple macro?